packages/runtime/src/lesson-files.ts

```diff
@@ -2,13 +2,41 @@ import type { Files, FilesRef, Lesson } from '@tutorialkit/types';
 import { newTask, type Task } from './tasks.js';
 import { wait } from './utils/promises.js';
 
+type InvalidationResult =
+  | {
+      type: 'template' | 'files' | 'solution';
+      files: Files;
+    }
+  | { type: 'none' };
+
 export class LessonFilesFetcher {
   private _map = new Map<string, Files>();
   private _templateLoadTask?: Task<Files>;
   private _templateLoaded: string | undefined;
 
+  async invalidate(fileRef: string): Promise<InvalidationResult> {
+    if (!this._map.has(fileRef)) {
+      return { type: 'none' };
+    }
+
+    const type = fileRef.startsWith('template-') ? 'template' : fileRef.endsWith('files.json') ? 'files' : 'solution';
+
+    let files: Files;
+
+    if (this._templateLoaded === fileRef) {
+      files = await this._fetchTemplate(fileRef).promise;
+    } else {
+      files = await this._fetchFiles(fileRef);
+    }
+
+    return {
+      type,
+      files,
+    };
+  }
+
   async getLessonTemplate(lesson: Lesson): Promise<Files> {
-    const templatePathname = `template-${lesson.data.template}`;
+    const templatePathname = `template-${lesson.data.template}.json`;
```

Comment:
Was this a bug? 👀 ?

Reply:
It wasn't but I've normalized the logic around `filesRef`. In particular:
* `filesRef` is a string that semantically points to a list of files. It has to be fetched on the client to resolve to those files through the `LessonFilesFetcher`.